nuxeo-mongodb-directory/src/main/java/org/nuxeo/directory/MongoDBSession.java

```
@@ -0,0 +1,349 @@
+        // a timeout will occur if a command is launched with bad username/password
+        MongoClient mongoClient = new MongoClient(client.getAddress(),
+                Arrays.asList(MongoCredential.createCredential(username, dbName, password.toCharArray())),
+                MongoClientOptions.builder().serverSelectionTimeout(1000).build());
+        try {
+            mongoClient.getDatabase(dbName).runCommand(new Document("ping", 1));
+            return true;
+        } catch (MongoTimeoutException e) {
+            return false;
+        } finally {
+            mongoClient.close();
+        }
+    }
+
+    @Override
+    public boolean hasEntry(String id) {
+        return getCollection().count(
+                MongoDBSerializationHelper.fieldMapToBson(Collections.singletonMap(getIdField(), id))) > 0;
+    }
+
+    @Override
+    public DocumentModel createEntry(DocumentModel documentModel) {
+        return createEntry(documentModel.getProperties(schemaName));
+    }
+
+    @Override
+    public DocumentModel getEntryFromSource(String id, boolean fetchReferences) throws DirectoryException {
+        return query(Collections.singletonMap(getIdField(), id), Collections.emptySet(), Collections.emptyMap(),
+                fetchReferences, 1, -1).get(0);
+
+    }
+
+    public MongoCollection<Document> getCollection() {
+        return MongoDBConnectionHelper.getCollection(client, dbName, directoryName);
+    }
+
+    public boolean hasCollection(String collection) {
+        return MongoDBConnectionHelper.hasCollection(client, dbName, collection);
+    }
+
+    protected DocumentModel fieldMapToDocumentModel(Map<String, Object> fieldMap) {
+        String id = String.valueOf(fieldMap.get(getIdField()));
+        try {
+            DocumentModel docModel = BaseSession.createEntryModel(null, schemaName, id, fieldMap, isReadOnly());
+            return docModel;
+        } catch (PropertyException e) {
```

As this is a new directory type let's remove this crappy exception swallowing and just let the `PropertyException` be raised to the caller.